### PR TITLE
[ch32722] ePD-amp changes and fixes

### PR DESCRIPTION
**Pull request by emaciupe**
[ch32722]
Includes: 
1. Add docstring to mapping - code review

2. Activate/Inactivate user when changing realms

3. Fix removing all groups - groups: []

4. Add Menu link in admin header

5. Add organization_id to api/audit/audit-firms/current_tenant/ response

### CODE REVIEW

File: src/etools/applications/audit/views.py [hunk 1]
```diff
@@ -149,7 +150,16 @@ def get_queryset(self):
         if self.request.query_params.get('verbosity', 'full') != 'minimal':
             queryset = queryset.select_related('profile__organization__auditorfirm')
 
-        return queryset
+        realm_context = {
+            "country": connection.tenant,
+            "organization_id__in": Organization.objects.filter(
+                vendor_number__in=['UNICEF', '000']).values_list('id', flat=True),
```
Comment (robertavram) [inline src/etools/applications/audit/views.py:156]: keep only UNICEF org

File: src/etools/applications/audit/views.py [hunk 2]
```diff
@@ -132,10 +133,10 @@ class AuditUsersViewSet(generics.ListAPIView):
     """
 
     permission_classes = (IsAuthenticated, )
-    filter_backends = (SearchFilter, DjangoFilterBackend, UnicefUsersAllowedFilter)
+    filter_backends = (SearchFilter, DjangoFilterBackend)
```
Comment (robertavram) [inline src/etools/applications/audit/views.py:136]: let's revisit!

File: src/etools/applications/tpm/tpmpartners/synchronizers.py [hunk 3]
```diff
@@ -54,6 +55,10 @@ def _partner_save(self, partner):
                 'hidden': True if partner['POSTING_BLOCK'] or partner['MARKED_FOR_DELETION'] else False,
             }
             partner, _ = TPMPartner.objects.update_or_create(organization=organization, defaults=defaults)
+
+            country = Country.objects.get(schema_name=defaults['country'] if defaults['country'] else connection.tenant.schema_name)
```
Comment (robertavram) [inline src/etools/applications/tpm/tpmpartners/synchronizers.py:59]: connection tenant schema

Comment (robertavram) [reply src/etools/applications/tpm/tpmpartners/synchronizers.py:59]: i think we need to set this somewhere else

File: src/etools/applications/tpm/tpmpartners/synchronizers.py [hunk 4]
```diff
@@ -93,9 +98,8 @@ def deactivate_staff_members(partner):
         # users_deactivate = User.objects.filter(tpmpartners_tpmpartnerstaffmember__in=staff_members)
         # users_deactivate.update(is_active=False)
         try:
-            country = Country.objects.get(name=partner.country)
             Realm.objects.filter(
-                country=country,
+                country__in=partner.countries.all(),
```
Comment (robertavram) [inline src/etools/applications/tpm/tpmpartners/synchronizers.py:102]: country not needed, deactivate all realms (the TPM active groups) in all countries that have this organization. remove country filter

File: src/etools/applications/users/serializers_v3.py [hunk 5]
```diff
@@ -241,21 +244,18 @@ def create(self, validated_data):
...
         self.create_realms(instance, organization_id, group_ids)
+        instance.update_active_state()
         notify_user_on_realm_update.delay(instance.id)
         return instance
 
 
 class UserRealmUpdateSerializer(UserRealmBaseSerializer):
     def update(self, instance, validated_data):
-        organization_id = validated_data.get('organization', instance.profile.organization.id)
+        organization_id = validated_data.get('organization', self.context['request'].user.profile.organization.id)
```
Comment (robertavram) [inline src/etools/applications/users/serializers_v3.py:254]: if you are a UNICEF User you need to send the organization id if you're not a unicef user, the organization should be taken from the user herself.

**Review by robertavram — CHANGES_REQUESTED** [review]

File: src/etools/applications/tpm/tpmpartners/synchronizers.py (hunk 3 above)
Comment (emaciupe) [reply src/etools/applications/tpm/tpmpartners/synchronizers.py:59]: @robertavram there is a dedicated endpoint on partners/id/activate/ that also sets the country

**Pull request merged**